src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -90,48 +77,47 @@ internal AuthenticationOptions BaseOptions
             }
         }
 
-        private static void OnSendingHeaderCallback(object state)
+        private static async Task OnResponseStartingCallback(object state)
```

Comment:
@Tratcher Is there a reason this is a static passing this in, instead of just making this an instance method with null state (soon to be using the stateless sugar method)?